dmutils/logging.py

```diff
@@ -77,16 +88,15 @@ def filter(self, record):
         return record
 
 
-class RequestIdFilter(logging.Filter):
-    @property
-    def request_id(self):
-        if has_request_context() and hasattr(request, 'request_id'):
-            return request.request_id
-        else:
-            return 'no-request-id'
-
+class RequestExtraContextFilter(logging.Filter):
+    """
+        Filter which will pull extra context from the current request's `get_extra_log_context` method (if present)
+        and make this available on log records
+    """
     def filter(self, record):
-        record.request_id = self.request_id
+        if has_request_context() and callable(getattr(request, "get_extra_log_context", None)):
+            for key, value in request.get_extra_log_context().items():
+                setattr(record, key, value or "-")
```

Comment:
Just checking my understanding - if the context value is `None` this will log as `-`? Would `'None'` or `'null'` be more useful for a dev looking through the logs?

😹 I just googled for other examples and someone literally wrote an academic paper on logging nulls: http://thesai.org/Downloads/Volume1No1/Paper_1-A_New_Approach_for_Handling_Null_Values_in_Web_Server_Log.pdf

Reply:
Well the reason I chose it is because it's the "standard" way of doing it for line-based logging, and wasn't really going to start hacking either of the formatters to have per-formatter null representation. But I'm not particularly tied to `-`, just wanted to terminate the `None`s *somewhere* lest something downstream isn't robust against it. Thoughts anyone?

Reply:
Actually, y'know, I *could* put the substitution in `CustomLogFormatter` and just let the nulls come through to the json output so that we had an easy way of seeing that there's no value. That might be better.